ui/src/main/java/edu/wpi/grip/ui/util/NodeControllerManager.java

```diff
@@ -0,0 +1,268 @@
+package edu.wpi.grip.ui.util;
+
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import edu.wpi.grip.ui.Controller;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.scene.Node;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Keeps track of the Nodes in an observable list and provides mapping from a {@link Node}
+ * to its respective {@link Controller}. This makes it easy to keep track of what controller
+ * is mapped to what node without having to use a secondary backing list with index comparisons.
+ * <p>
+ * This utility also provides helpful functionality like sorting the nodes in the list using a
+ * comparator.
+ *
+ * @param <C> The Controller type for ever node in this list
+ * @param <N> The type of the Node being mapped to a Controller.
+ */
+public final class NodeControllerManager<C extends Controller, N extends Node> implements Map<C, N> {
```

Comment:
I think only `put`, `get`, and `keySet` are used.  We should not have like 20 methods we don't use